client/src/components/graph/graph.js

```diff
@@ -404,7 +404,11 @@ class Graph extends React.Component {
 
     /* clear out whatever was on the div, even if nothing, but usually the brushes etc */
 
-    d3.select("#lasso-layer").selectAll(".lasso-group").remove();
+    const lassoGroupElem = d3.select("#lasso-layer").selectAll(".lasso-group");
```

Comment:
I would just rewrite all of these lines as a single line:   `d3.select("#lasso-layer")?.selectAll(".lasso-group")?.remove();`

note the double optional chaining operators